src/distributions/bernoulli.rs

```diff
@@ -39,7 +39,7 @@ impl Bernoulli {
 impl Distribution<bool> for Bernoulli {
     #[inline]
     fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> bool {
-        rng.gen_bool(self.p)
+        rng.gen::<f64>() <= self.p
```

Comment:
I guess this is a reasonable implementation (with our open range sampling). The idea here is to get 52-bits of precision instead of 32 I guess?